pkg/plugins/handlers.go

```diff
@@ -17,15 +17,42 @@ import (
 type PluginsHandler struct {
 	Client             *http.Client
 	PluginsEndpointMap map[string]string
+	PublicDir          string
 }
 
-func NewPluginsHandler(client *http.Client, token string, pluginsEndpointMap map[string]string) *PluginsHandler {
+func NewPluginsHandler(client *http.Client, pluginsEndpointMap map[string]string, publicDir string) *PluginsHandler {
 	return &PluginsHandler{
 		Client:             client,
 		PluginsEndpointMap: pluginsEndpointMap,
+		PublicDir:          publicDir,
 	}
 }
 
+func (p *PluginsHandler) HandleLocales(w http.ResponseWriter, r *http.Request) {
+	if r.Method != "GET" {
+		w.Header().Set("Allow", "GET")
+		serverutils.SendResponse(w, http.StatusMethodNotAllowed, serverutils.ApiError{Err: "Method unsupported, the only supported methods is GET"})
+		return
+	}
+
+	query := r.URL.Query()
+	lang := query.Get("lng")
+	// In case of static plugin the localization namespace should contain name of the json file that contains the localization strings,
+	// since various static plugins have different names for the localization file.
+	// eg. namespace query for the local-storage-operator-plugin locales - '&ns=lso-plugin.json'
+	// In case of the dynamic plugin the localization namespace should contain name of the plugin prefixed with 'plugin:' prefix
+	// No file extension is needed since for the dynamic plugins we will fetch 'default.json' from the plugin pod.
+	// eg. 'plugin:helm-plugin'
+	pluginLocale := query.Get("ns")
+	if !strings.HasPrefix(pluginLocale, "plugin:") {
+		http.ServeFile(w, r, path.Join(p.PublicDir, "locales", lang, pluginLocale))
+		return
+	}
+	// in case of dynamic-plugin we need to trim the "plugin:" prefix.
+	pluginName := strings.TrimPrefix(pluginLocale, "plugin:")
+	http.Redirect(w, r, path.Join(r.Host, "api/plugins", pluginName, "locales", lang, "default.json"), http.StatusSeeOther)
```

Comment:
We should confirm that i18next-http-backend handles redirects. It might be better just to proxy the content here if it's not too hard.

Reply:
yeah, proxying makes more sense ... updated